File: Frontend_Flutter/lib/models/order.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'order_item.dart';

part 'order.freezed.dart';
part 'order.g.dart';

@freezed
class Order with _$Order {
  const factory Order({
    int? id,
    int? userId,
    String? userName,
    int? restaurantId,
    String? restaurantName,
    String? status,
    double? totalPrice,
    double? finalPrice,
    DateTime? createdAt,
    @Default([]) List<OrderItem> items,
    // Payment fields
    String? paymentMethod,
    String? paymentStatus,
    String? shippingAddress,
    String? phoneNumber,
    String? note,
    double? shippingFee,
    double? discountAmount,
    String? voucherCode,
  }) = _Order;

  factory Order.fromJson(Map<String, dynamic> json) => _$OrderFromJson(json);
}

File: Backend-SpringBoot/FoodApp/src/main/java/com/tientien/foodapp/order/entity/Order.java
package com.tientien.foodapp.order.entity;

import com.tientien.foodapp.common.entity.BaseEntity;
import com.tientien.foodapp.common.enums.OrderStatus;
import com.tientien.foodapp.restaurant.entity.Restaurant;
import com.tientien.foodapp.user.entity.User;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Entity
@Table(name = "orders")
@Getter
@Setter
public class Order extends BaseEntity {

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "restaurant_id")
    private Restaurant restaurant;

    @Enumerated(EnumType.STRING)
    private OrderStatus status;

    private Double totalPrice;
    private Double finalPrice;

    @OneToMany(
            mappedBy = "order",
            cascade = CascadeType.ALL
    )
    private List<OrderItem> items;

    // Payment fields
    private String paymentMethod; // COD, VNPAY, MOMO, CARD
    private String paymentStatus; // PENDING, PAID, FAILED, REFUNDED
    private String shippingAddress;
    private String phoneNumber;
    private String note;
    private Double shippingFee;
    private Double discountAmount;
    private String voucherCode;
}
